evm/src/main/java/org/hyperledger/besu/evm/precompile/BLAKE2BFPrecompileContract.java

```diff
@@ -77,6 +84,42 @@ public PrecompileContractResult computePrecompile(
       return PrecompileContractResult.halt(
           null, Optional.of(ExceptionalHaltReason.PRECOMPILE_ERROR));
     }
-    return PrecompileContractResult.success(Hash.blake2bf(input));
+    PrecompileInputResultTuple res = null;
+    Integer cacheKey = null;
+    if (enableResultCaching) {
+      cacheKey = getCacheKey(input);
+      res = blakeCache.getIfPresent(cacheKey);
+      if (res != null) {
+        if (res.cachedInput().equals(input)) {
+          cacheEventConsumer.accept(new CacheEvent(PRECOMPILE_NAME, CacheMetric.HIT));
+          return res.cachedResult();
+        } else {
+          LOG.info(
+              "false positive blake2bf {}, cache key {}, cached input: {}, input: {}",
+              input.getClass().getSimpleName(),
+              cacheKey,
+              res.cachedInput().toHexString(),
+              input.toHexString());
+
+          cacheEventConsumer.accept(new CacheEvent(PRECOMPILE_NAME, CacheMetric.FALSE_POSITIVE));
+        }
+      } else {
+        cacheEventConsumer.accept(new CacheEvent(PRECOMPILE_NAME, CacheMetric.MISS));
+      }
+    }
+
+    res =
+        new PrecompileInputResultTuple(
+            input.copy(), PrecompileContractResult.success(Hash.blake2bf(input)));
```

Comment:
I think it is still worth checking if enableResultCaching is enabled before doing the copy.